docs/polygon_draw.js

```diff
@@ -366,22 +387,79 @@ function setUpPolygonDrawing(map, firebasePromise) {
  * @param {google.maps.Map} map Map to display regions on
  * @param {Promise<any>} firebasePromise Promise with Firebase damage data (also
  *     implies that authentication is complete)
- * @return {Promise} promise that is resolved when all initialization is done
- *     (only used by tests)
+ * @return {?google.maps.drawing.DrawingManager} DrawingManager added to map
+ *     (only used by tests). Null if there was an error
  */
-function initializeAndProcessUserRegions(map, firebasePromise) {
+async function initializeAndProcessUserRegions(map, firebasePromise) {
   addLoadingElement(mapContainerId);
-  return firebasePromise
-      .then((doc) => {
-        // Damage asset may not exist yet, so this is undefined. We tolerate
-        // gracefully.
-        damageAsset = doc.data()['asset_data']['damage_asset_path'];
-        userShapes = getFirestoreRoot().collection(collectionName);
-      })
-      .then(() => userShapes.get())
-      .then(
-          (querySnapshot) => drawRegionsFromFirestoreQuery(querySnapshot, map))
-      .catch(createError('getting user-drawn regions'));
+  try {
+    setUpPopup();
```

Comment:
Does all of this need to be in this try block? Like can setUpPopup fail?

Reply:
I don't think there's any harm in it. Basically, no matter what, we should make sure loadingElementFinished(mapContainerId) is called. Wrapping everything in the try means there's also no chance of programmer error, for instance if there's an early return.

Reply:
Can you say more about why that makes debugging easier? This try block doesn't catch anything, so there won't be any ambiguous messages about a thrown exception. I swapped setUpPopup() and addLoadingElement so the issue about unexpected exceptions leaving us vulnerable to loading forever should be moot.

Reply:
Ah so you're just doing it this way to make sure whatever gets returned we do the finish loading stuff? Got it.